test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -130,77 +130,67 @@ public String getID() {
 
-            final Map<String, Object> pravegaSpec = ImmutableMap.<String, Object>builder().put("controllerReplicas", controllerCount)
-                    .put("segmentStoreReplicas", segmentStoreCount)
-                    .put("debugLogging", true)
-                    .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                    .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
-                    .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
-                    .put("options", props)
-                    .put("image", getImageSpec(pravegaImg, PRAVEGA_VERSION))
-                    .put("tier2", tier2Spec())
-                    .build();
+        }
+
+        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder().put("image", bookkeeperImgSpec)
+                .put("replicas", bookieCount)
+                .put("resources", getResources("2000m", "5Gi", "1000m", "3Gi"))
+                .put("storage", ImmutableMap.builder()
+                        .put("indexVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .put("ledgerVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .put("journalVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .build())
+                .put("autoRecovery", true)
+                .build();
+
+        final Map<String, Object> pravegaSpec = ImmutableMap.<String, Object>builder().put("controllerReplicas", controllerCount)
+                .put("segmentStoreReplicas", segmentStoreCount)
+                .put("debugLogging", true)
+                .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
+                .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
+                .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
+                .put("options", props)
+                .put("image", pravegaImgSpec)
+                .put("tier2", tier2Spec())
+                .build();
 
+        return ImmutableMap.<String, Object>builder()
+                .put("apiVersion", "pravega.pravega.io/v1alpha1")
+                .put("kind", CUSTOM_RESOURCE_KIND_PRAVEGA)
+                .put("metadata", ImmutableMap.of("name", PRAVEGA_ID, "namespace", NAMESPACE))
+                .put("spec", pravegaClusterSpecBuilder(zkLocation, bookkeeperSpec, pravegaSpec))
+                .build();
+    }
+
+    protected Map<String, Object> pravegaClusterSpecBuilder(String zkLocation, Map<String, Object> bookkeeperSpec, Map<String, Object> pravegaSpec) {
+        if (isLaterOperator) {
+            return ImmutableMap.<String, Object>builder()
+                    .put("zookeeperUri", zkLocation)
+                    .put("bookkeeper", bookkeeperSpec)
+                    .put("pravega", pravegaSpec)
+                    .put("version", PRAVEGA_VERSION)
```

Comment:
If adding the version is the only additional step then we can refactor the code further.

Reply:
done